apps/aecore/test/aec_block_fork_tests.erl

```diff
@@ -0,0 +1,103 @@
+-module(aec_block_fork_tests).
+
+-include_lib("eunit/include/eunit.hrl").
+
+%%%===================================================================
+%%% Test cases
+%%%===================================================================
+
+apply_minerva_test_() ->
+    {foreach,
+     fun() ->
+         meck:new(aec_fork_block_settings, [passthrough]),
+         ok
+     end,
+     fun(ok) ->
+         meck:unload(aec_fork_block_settings),
+         ok
+     end,
+     [ {"Empty minerva migration does not change balances",
+        fun() ->
+            InitialAccounts = 
+                [{Alice, BalA},
+                 {Bob,   BalB},
+                 {Carol, BalC}] = generate_accounts(3),
+            T0 = make_trees(InitialAccounts), % Alice, Bob and Carol
+            meck_minerva_accounts([]), % no accounts migrated
+            T1 = aec_block_fork:apply_minerva(T0),
+            assert_only_accounts_tree_changed(T0, T1),
+            assert_balance(T1, Alice, BalA),
+            assert_balance(T1, Bob,   BalB),
+            assert_balance(T1, Carol, BalC),
+            ok
+        end},
+       {"Minerva migration changes balances",
+        fun() ->
+            [{Alice,  BalA} = A,
+              {Bob,   BalB} = B,
+              {Carol, BalC}] = generate_accounts(3),
+            T0 = make_trees([A, B]), % only Alice and Bob are present pre-minerva
+            meck_minerva_accounts([{Alice, DeltaA = 10}, % Alice had migrared more
+                                   {Carol, BalC}]), % Carol is new
+            T1 = aec_block_fork:apply_minerva(T0),
+            assert_only_accounts_tree_changed(T0, T1),
+            assert_balance(T1, Alice, BalA + DeltaA), % Alice balance is increased
+            assert_balance(T1, Bob,   BalB), % Bob is unchanged
+            assert_balance(T1, Carol, BalC), % Carol is inserted
+            ok
+        end}
+     ]}.
+
+meck_minerva_accounts(AccountsList) ->
+    meck:expect(aec_fork_block_settings, minerva_accounts,
+                fun() -> AccountsList end).
+
+assert_only_accounts_tree_changed(Trees0, Trees1) ->
```

Comment:
I would be tempted to add this check in `aec_trees:perform_pre_transformation` (when at hard_fork height), but maybe that is overkill...

Reply:
It could be nice to make util functions for this code at least. To encourage checking properties based on in which tree changes occur.

Reply:
But not needed for this PR.